src/Symfony/Component/PropertyAccess/PropertyPathInterface.php

```diff
@@ -83,4 +83,15 @@ public function isProperty(int $index);
      * @throws Exception\OutOfBoundsException If the offset is invalid
      */
     public function isIndex(int $index);
+
+    /**
+     * Returns whether the element at the given index is optional.
+     *
+     * @param int $index The index in the property path
+     *
+     * @return bool Whether the element at this index is an array index
+     *
+     * @throws Exception\OutOfBoundsException If the offset is invalid
+     */
+    public function isOptional(int $index);
```

Comment:
Missing return type.
Note that adding a new method on an existing interface is a BC break.
We use `@method` annotations in similar cases. See branch 4.4 for examples.

Reply:
Did it, 

just when I am using it shouldn't I trigger a deprecation if the method is not set. 
I found this example in the 4.4 branch and it's not present: https://github.com/symfony/symfony/blob/4.4/src/Symfony/Bundle/SecurityBundle/DataCollector/SecurityDataCollector.php#L120